portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,89 +20,126 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
 import com.google.gson.Gson;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime; 
+import java.lang.Math;
+import com.google.common.base.Strings;
 
-/** Servlet that handles comment data. */
+
+/** 
+  * Servlet that uploads and retrieves persistent comment data using datastore.
+  */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  /** Represents a single comment */
+  private static class Comment {
+      String text;
+      String name;
+      long time;
+
+      Comment(String text, String name, long time) {
+        this.text = text;
+        this.name = name;
+        this.time = time;
+      }
+
+      /** Returns an entity representing this comment. */
+      static Entity toEntity(Comment c) {
```

Comment:
I can see where you're coming from, however this method doesn't need to be static. Here's why: 
If you need to convert an instance, you have it already, so you can just use c.toEntity(). save yourself some wasted characters on the static aspect :)  

Now, if this were on the Entity class, called "fromComment", that would make sense.